### PR TITLE
[language] More command-line-common cleanup

### PR DESCRIPTION
**Pull request by tnowacki**
- Migrated move_stdlib::utils into command-line-common
- Cleaned up some cloning around IR compiler interface

## Motivation

- Consolidation of common practices in language/ makes things a bit more maintainable 

## Test Plan

- Ran tests

### CODE REVIEW

**Review by modocache — APPROVED** [review]
Cool, looks good to me -- thanks!

File: language/move-command-line-common/src/files.rs [hunk 1]
```diff
@@ -19,15 +19,18 @@ pub const MOVE_ERROR_DESC_EXTENSION: &str = "errmap";
...
 
     for s in paths {
-        let path = Path::new(s);
+        let path = s.as_ref();
         if !path.exists() {
-            return Err(anyhow!("No such file or directory '{}'", s));
+            return Err(anyhow!(
+                "No such file or directory '{}'",
+                path.to_string_lossy()
+            ));
```
Comment (tzakian) [inline language/move-command-line-common/src/files.rs:33]: Nit: `bail!` here would be more concise

File: language/tools/move-cli/src/sandbox/utils/mod.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ pub use mode::*;
 pub use on_disk_state_view::*;
 pub use package::*;
 
-pub(crate) fn get_gas_status(gas_budget: Option<u64>) -> Result<GasStatus<'static>> {
+pub fn get_gas_status(gas_budget: Option<u64>) -> Result<GasStatus<'static>> {
```
Comment (tzakian) [inline language/tools/move-cli/src/sandbox/utils/mod.rs:38]: Do we want to keep this crate public still? I don't see a reason for it to be public/it being used anywhere.

**Review by tzakian — APPROVED** [review]
LGTM. Two nits, but loving these cleanups :)

File: language/tools/move-cli/src/sandbox/utils/mod.rs (hunk 2 above)
Comment (tnowacki) [reply language/tools/move-cli/src/sandbox/utils/mod.rs:38]: I think I ended up using it in some testing code I was writing. But I can revert the change to that PR

**Review by tzakian — APPROVED** [review]

**Pull request closed without merging**